### PR TITLE
Use `<link />` on Dropdown

### PR DESCRIPTION
**Pull request by dani-moreno**
Adapts the Dropdown component to use the `Link` component for links.

### CODE REVIEW

File: lib/experimental/Navigation/Dropdown/index.tsx [hunk 1]
```diff
@@ -79,9 +98,7 @@ export function Dropdown({ items, children }: DropdownProps) {
               key={index}
               item={{
                 ...item,
-                onClick: () => {
-                  item.onClick()
-                },
+                onClick: item.onClick ? () => item.onClick?.() : undefined,
```
Comment (RamProg) [inline lib/experimental/Navigation/Dropdown/index.tsx:101]: This looks redundant, you are checking twice for item.onClick

Comment (RamProg) [reply lib/experimental/Navigation/Dropdown/index.tsx:101]:
```suggestion
                onClick: () => item.onClick?.()
```

Comment (dani-moreno) [reply lib/experimental/Navigation/Dropdown/index.tsx]: Thanks! Can't it be just `onClick: item.onClick`, as it is not defined, it's going to be `undefined` anyway? Or am I missing something?

Comment (RamProg) [reply lib/experimental/Navigation/Dropdown/index.tsx]: Yep, that should work too

Comment (RamProg) [reply lib/experimental/Navigation/Dropdown/index.tsx]: What wouldn't work is the original because if onClick is undefined it will break

**Review by RamProg — APPROVED** [review]

**Pull request merged**